src/test/java/org/javarosa/xpath/expr/XPathFuncExprBase64DecodeTest.java

```diff
@@ -0,0 +1,97 @@
+package org.javarosa.xpath.expr;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.javarosa.core.test.AnswerDataMatchers.stringAnswer;
+import static org.javarosa.core.util.BindBuilderXFormsElement.bind;
+import static org.javarosa.core.util.XFormsElement.body;
+import static org.javarosa.core.util.XFormsElement.head;
+import static org.javarosa.core.util.XFormsElement.html;
+import static org.javarosa.core.util.XFormsElement.input;
+import static org.javarosa.core.util.XFormsElement.mainInstance;
+import static org.javarosa.core.util.XFormsElement.model;
+import static org.javarosa.core.util.XFormsElement.t;
+import static org.javarosa.core.util.XFormsElement.title;
+
+import java.io.IOException;
+import java.util.Arrays;
+import org.javarosa.core.test.Scenario;
+import org.javarosa.xform.parse.XFormParser;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Enclosed.class)
```

Comment:
I wanted to have both valid and invalid input cases in the same file. This is the cleanest way I found to do that.

Reply:
I agree with you in some cases but I think it really depends on the shape of the tests. With this style of test where there's a standard form with different values subbed in, I find parameterized tests easier to work with. We have a lot of examples of `Parameterized` tests in JR.

That said, I'm ok moving to lowercase-p parameterized tests. I've done that in the latest commit. I think the important thing is that the form is only defined once. Hopefully this is a reasonable compromise for you.

Reply:
I much prefer the new approach. I find using common builders/helpers a much nicer approach than an entire parametrized test, as you're still able to have a unique procedure for each case (without branching).

Reply:
Makes sense! We can move towards that as we introduce new tests.